cmd/tuf/remove_key.go

```diff
@@ -0,0 +1,42 @@
+package main
+
+import (
+	"os"
+	"path/filepath"
+
+	"github.com/theupdateframework/go-tuf"
+	"github.com/flynn/go-docopt"
+)
+
+func init() {
+	register("remove-key", cmdRemoveKey, `
+usage: tuf remove-key [--expires=<days>] <role> <id>
+
+Remove a signing key
+
+Before the key is removed the key will be first revoked
+The key will then be removed from the root metadata file and if the key is present in 
+"keys" directory it will also be removed
+
+Options:
+  --expires=<days>   Set the root metadata file to expire <days> days from now.
+`)
+}
+
+func cmdRemoveKey(args *docopt.Args, repo *tuf.Repo) error {
+	role := args.String["<roles>"]
+	keyID := args.String["<id>"]
+
+	err := cmdRevokeKey(args, repo)
+	if err != nil {
+		return err
+	}
+
+	keyPath := filepath.Join("keys", keyID)
+	if _, err := os.Stat(keyPath); err == nil {
+		if err := os.Remove(keyPath); err != nil {
+			return err
+		}
+	}
```

Comment:
Ah, so I realized that this part is going to be a bit more difficult than I thought. The revoke part is clear though!

Basically, keys are indeed stored in the `keys` directory, but a key is stored (encrypted or in plain text) in a <role>.json file so it's not going to be possible to remove it with the suggested code :/ 

In practice, removing a key would translate to:
- unmarshal that <role>.json file storing the key
- possibly decrypt the key
- delete the given key (note you'd have to find first which key entry corresponds to that keyid as there are no keyids there)
- on that note, it's not safe to assume that you're running the tuf cli from the same directory where those files and folders are, instead it would be better to use the dir variable storing the location of the filestore 👍 

Try running `./tuf init` and then `./tuf gen-key root' a few times (with a passphrase or just enter to skip it) so you can get a better idea of what I want to say 👍

Reply:
Okay yes, I see what you are saying and I think this will be a bit of a task for me since I believe I have to decrypt the key if it is encrypted. So it might take me some time to fully get a solution. I hope I may reach out if i have any question or I am a bit stuck

Reply:
Of course! Have a look at `local_store.go` where most of the encrypting/decrypting happens to get a sense of the workflow 👍